### PR TITLE
fix: default mode in the light/dark mode toggle is light

### LINKED ISSUE #426: [BUG] The default mode in the light/dark mode toggle is light
### Description

The web site default mode is dark but in the light/dark mode toggle is light

### Screenshots

image

### Additional information

_No response_

### 🥦 Browser

Google Chrome

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by itsankitbhusal**
# Fixes Issue

**My PR closes #426 **

# 👨‍💻 Changes proposed(What did you do ?)
- Changed optional chaining `??` to ternary operator `?:` which sets the dark mode to false on default when not found on local storage and changes accordingly afterward.

# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots
image
Screenshot from 2023-10-15 20-22-12

### CODE REVIEW

File: client/src/context/DarkModeContext.jsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ export function useDarkMode() {
 
 export function DarkModeProvider({ children }) {
 
-    const [darkMode, setDarkMode] = useState(() => JSON.parse(window.localStorage.getItem("darkMode")) ?? true);
+    const [darkMode, setDarkMode] = useState(() => (JSON.parse(window.localStorage.getItem("darkMode")) ? true: false));
```
Comment (Dun-sin) [inline client/src/context/DarkModeContext.jsx:12]:
```suggestion
    const defaultMode = JSON.parse(window.localStorage.getItem("darkMode")
    const [darkMode, setDarkMode] = useState(() => (defaultMode === undefined || defaultMode === null) ? true : defaultMode );
```

File: client/src/context/DarkModeContext.jsx [hunk 2]
```diff
@@ -9,7 +9,8 @@ export function useDarkMode() {
 
 export function DarkModeProvider({ children }) {
 
-    const [darkMode, setDarkMode] = useState(() => JSON.parse(window.localStorage.getItem("darkMode")) ?? true);
+    const defaultMode = JSON.parse(window.localStorage.getItem("darkMode")
+    const [darkMode, setDarkMode] = useState(() => (defaultMode === undefined || defaultMode === null) ? true : defaultMode );
```
Comment (Dun-sin) [inline client/src/context/DarkModeContext.jsx:13]:
```suggestion
    const [darkMode, setDarkMode] = useState(() => JSON.parse(window.localStorage.getItem("darkMode")) ?? true);
```

File: client/src/context/DarkModeContext.jsx [hunk 3]
```diff
@@ -8,8 +8,8 @@ export function useDarkMode() {
 }
 
 export function DarkModeProvider({ children }) {
-
-    const [darkMode, setDarkMode] = useState(() => JSON.parse(window.localStorage.getItem("darkMode")) ?? true);
+	  const defaultMode = JSON.parse(window.localStorage.getItem("darkMode");
```
Comment (Dun-sin) [inline client/src/context/DarkModeContext.jsx:11]:
```suggestion
	  const defaultMode = JSON.parse(window.localStorage.getItem("darkMode"));
```

File: client/src/context/DarkModeContext.jsx [hunk 4]
```diff
@@ -8,8 +8,8 @@ export function useDarkMode() {
 }
 
 export function DarkModeProvider({ children }) {
-
-    const [darkMode, setDarkMode] = useState(() => JSON.parse(window.localStorage.getItem("darkMode")) ?? true);
+	  const defaultMode = JSON.parse(window.localStorage.getItem("darkMode"));
+	  const [darkMode, setDarkMode] = useState(() => (defaultMode === undefined || defaultMode === null) ? true : defaultMode );
```
Comment (Dun-sin) [inline client/src/context/DarkModeContext.jsx:12]:
```suggestion
    const defaultMode = JSON.parse(window.localStorage.getItem("darkMode"));
    const [darkMode, setDarkMode] = useState(defaultMode !== undefined ? defaultMode : true);
```

File: client/src/context/DarkModeContext.jsx [hunk 5]
```diff
@@ -8,8 +8,8 @@ export function useDarkMode() {
 }
 
 export function DarkModeProvider({ children }) {
-
-    const [darkMode, setDarkMode] = useState(() => JSON.parse(window.localStorage.getItem("darkMode")) ?? true);
+    const defaultMode = JSON.parse(window.localStorage.getItem("darkMode"));
+    const [darkMode, setDarkMode] = useState(defaultMode !== undefined ? defaultMode : true);
```
Comment (Dun-sin) [inline client/src/context/DarkModeContext.jsx:12]:
```suggestion
    const [darkMode, setDarkMode] = useState(defaultMode !== null ? defaultMode : true);
```

**Comment by Dun-sin** [comment]
@itsankitbhusal this doesn't solve the problem and ruined a change, which i tried to fix lol, you have to fix the actual issue which is default mode toggle should be correct

**Comment by itsankitbhusal** [comment]
@Dun-sin have you fixed it already or is there still the changes necessary, if any mention it.
Changing from optional chaining to ternary worked on my machine on chrome, Firefox and brave.

**Comment by Dun-sin** [comment]
> @Dun-sin have you fixed it already or is there still the changes necessary, if any mention it. Changing from optional chaining to ternary worked on my machine on chrome, Firefox and brave.

what do you think this issue is meant to fix? i don't think you know what it's meant to fix

**Comment by itsankitbhusal** [comment]
@Dun-sin made changes and are pushed, have a look at it and comment if there is anything that needs to be changed.

**Pull request closed without merging**